OpenOversight/app/models.py

```diff
@@ -37,6 +37,22 @@ def __repr__(self):
         return '<Department ID {}: {}>'.format(self.id, self.name)
 
 
+class Rank(db.Model):
+    __tablename__ = 'ranks'
+
+    id = db.Column(db.Integer, primary_key=True)
+    rank = db.Column(db.String(255), index=True, unique=False, nullable=False)
```

Comment:
hmm, shouldn't there be a relationship between the rank and assignments tables (e.g. a rank added there should point to a column here)? i see that you're keeping the `rank` field in the assignments table as a separate field. is this preserved for the jobs vs ranks distinction you mention in the PR comments? If so, I think it will behoove us to have a generic table like `Jobs` and we could have a field in there that indicates whether or not it is a "officer rank", i.e. `is_sworn_officer`. what do you think? let me know if i misunderstand the purpose here

Reply:
Yup, that's basically the idea. The entries in this table basically just populate the rank dropdowns in the find and browse forms, while the `rank` field in the assignments table would list the job title, whether it was a "rank" or not. 

As for generalizing the table to `Jobs` with a `is_sworn_officer` field, I guess I just don't think it's necessary since the jobs that aren't sworn officer ranks aren't used in any forms or queries, and are already in the assignments table in the `rank` field. Are there other things you think a jobs table could be used for?

Reply:
Re-reading the above I think what I said was unclear: I was suggesting _instead_ of a `Ranks` table, we have a `Jobs` (or `EmployeeRole` or something) table, and the `is_sworn_officer` field in that field would determine whether or not that role should appear in the rank dropdowns (we don't need the `is_sworn_officer` flag for openoversight.com but I think y'all do for Baltimore hence this suggestion). 

The reason I'm advocating this is because there is duplication of data in the assignments table, which will complicate maintainability / extensibility (violates [2NF of database normalization](https://en.wikipedia.org/wiki/Second_normal_form)). For example, the assignments table and ranks/jobs table linked via foreign key would be useful:
* if we add data to the rank that we want to easily access on the officer's profile. Some possibilities are using a picture of the rank insignia, a description of what tasks that rank tends to do (maybe available in a tooltip), typical salary range of the rank (e.g. if at some later date we wanted to display "officer is 10% above the average" on their profile)))
* if for example a rank is renamed, e.g. we want to rename all `PO`s to `Police Officer` (that is exactly [this situation](https://github.com/lucyparsons/OpenOversight/pull/619/files#r257579546) which we have on openoversight.com due to the initial bad denormalized design which now requires a migration to address)

What do you think?

Reply:
Yup that makes sense and I agree that's a good way to go. I'll refactor.

Reply:
btw we should have `is_sworn_officer` be default True